src/Microsoft.TestPlatform.CrossPlatEngine/Client/InProcessProxyexecutionManager.cs

```diff
@@ -0,0 +1,126 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Client
+{
+    using System;
+    using System.Collections.ObjectModel;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+    using Microsoft.VisualStudio.TestPlatform.ObjectModel.Client;
+    using Microsoft.VisualStudio.TestPlatform.ObjectModel.Engine;
+    using Microsoft.VisualStudio.TestPlatform.ObjectModel.Engine.ClientProtocol;
+    using Microsoft.VisualStudio.TestPlatform.ObjectModel.Engine.TesthostProtocol;
+    using Microsoft.VisualStudio.TestPlatform.ObjectModel.Logging;
+
+    internal class InProcessProxyExecutionManager : IProxyExecutionManager
+    {
+        private ITestHostManagerFactory testHostManagerFactory;
+        IExecutionManager executionManager;
+        public bool IsInitialized { get; private set; } = false;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="InProcessProxyexecutionManager"/> class.
+        /// </summary>
+        public InProcessProxyExecutionManager() : this(new TestHostManagerFactory())
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="InProcessProxyexecutionManager"/> class.
+        /// </summary>
+        /// <param name="testHostManagerFactory">
+        /// Manager factory
+        /// </param>
+        internal InProcessProxyExecutionManager(ITestHostManagerFactory testHostManagerFactory)
+        {
+            this.testHostManagerFactory = testHostManagerFactory;
+            this.executionManager = this.testHostManagerFactory.GetExecutionManager();
+        }
+
+        /// <summary>
+        /// Initialize adapters.
+        /// </summary>
+        public void Initialize()
+        {
+            if (!this.IsInitialized)
+            {
+                // We don't need to pass list of extension as we are running inside vstest.console and
+                // it will use TestPluginCache of vstest.console
+                executionManager.Initialize(Enumerable.Empty<string>());
+                this.IsInitialized = true;
+            }
+        }
+
+        /// <inheritdoc/>
+        public int StartTestRun(TestRunCriteria testRunCriteria, ITestRunEventsHandler eventHandler)
+        {
+            try
+            {
+                // Initialize extension before execution if not already initialized
+                this.Initialize();
+
+                var executionContext = new TestExecutionContext(
+                            testRunCriteria.FrequencyOfRunStatsChangeEvent,
+                            testRunCriteria.RunStatsChangeEventTimeout,
+                            inIsolation: false,
```

Comment:
How is this used in TPv1? Are we missing scenario?

Reply:
This info get passed to adapter through RunContext